src/templating/templateEngine.js

```diff
@@ -37,11 +37,63 @@ ko.templateEngine.prototype['createJavaScriptEvaluatorBlock'] = function (script
 ko.templateEngine.prototype['makeTemplateSource'] = function(template, templateDocument) {
     // Named template
     if (typeof template == "string") {
+        var namedTemplateCache = this.namedTemplateCache || {};
+
+        // Cache entries last for 10s after the last access
+        var cacheLifetime = 10000;
+
+        // Registers a timer to clean up stale entries from the template cache.
+        function cleanupCache() {
```

Comment:
No good reason, I guess we could have it outside and then pass the `namedTemplateCache` as an argument?